var/spack/repos/builtin/packages/geomodel/package.py

```diff
@@ -54,6 +54,14 @@ class Geomodel(CMakePackage):
         when="+fullsimlight",
     )
 
+    variant(
+        "cxxstd",
+        default="17",
+        values=("11", "17", "20", "23"),
```

Comment:
(Red teaming hat on)  Is `geomodel cxxstd=11 +geomodelg4 ^geant4 cxxstd=17` expected to work?

Reply:
That's a good question. If they don't bake any standard assumptions into compiled code (like ROOT likes to do) and correctly guard standard constructs behind preprocessor flags, then I think yes. 

The more interesting question to me is: `cxxstd` is more of a loose convention for C++ based packages in Spack right? I've seen packages doing very different things with this variant. 

What is the recommended way to handle this?

Reply:
`cxxstd` is indeed more of a loose convention than anything defined properly in Spack. Reason for this is that _in an ideal world_ the C++ version of a dependency _shouldn't_ matter. In really (especially when we talk about using ROOT) it does, so there's a whole cluster of Spack packages which all depend on each other which must have the same C++ version. It's a bit unfortunate.

Reply:
Note, by the way, that you can construct dependencies in part. So:

```python
depends("root @6.21.0: cxxstd=20")
```

can be written as:

```python
depends("root @6.21.0:")
depends("root cxxstd=20")
```

So relatively complex dependencies (with many variants) can be split out into different statements. For ways in which this can be used, see e.g.:

* Celeritas: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/celeritas/package.py#L125
* ACTS (you may have heard of it): https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/acts/package.py#L417
* algebra-plugins: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/acts-algebra-plugins/package.py#L46